### ISSUE TITLE
two lxd processes running. Believed result of deb to snap conversion.

### ISSUE THREAD

**Issue by smoser** [lxc/lxd#4203]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04
 * The output of "lxc info":

        config: {}
        api_extensions:
        - storage_zfs_remove_snapshots
        - container_host_shutdown_timeout
        - container_stop_priority
        - container_syscall_filtering
        - auth_pki
        - container_last_used_at
        - etag
        - patch
        - usb_devices
        - https_allowed_credentials
        - image_compression_algorithm
        - directory_manipulation
        - container_cpu_time
        - storage_zfs_use_refquota
        - storage_lvm_mount_options
        - network
        - profile_usedby
        - container_push
        - container_exec_recording
        - certificate_update
        - container_exec_signal_handling
        - gpu_devices
        - container_image_properties
        - migration_progress
        - id_map
        - network_firewall_filtering
        - network_routes
        - storage
        - file_delete
        - file_append
        - network_dhcp_expiry
        - storage_lvm_vg_rename
        - storage_lvm_thinpool_rename
        - network_vlan
        - image_create_aliases
        - container_stateless_copy
        - container_only_migration
        - storage_zfs_clone_copy
        - unix_device_rename
        - storage_lvm_use_thinpool
        - storage_rsync_bwlimit
        - network_vxlan_interface
        - storage_btrfs_mount_options
        - entity_description
        - image_force_refresh
        - storage_lvm_lv_resizing
        - id_map_base
        - file_symlinks
        - container_push_target
        - network_vlan_physical
        - storage_images_delete
        - container_edit_metadata
        - container_snapshot_stateful_migration
        - storage_driver_ceph
        - storage_ceph_user_name
        - resource_limits
        - storage_volatile_initial_source
        - storage_ceph_force_osd_reuse
        - storage_block_filesystem_btrfs
        - resources
        - kernel_limits
        - storage_api_volume_rename
        - macaroon_authentication
        - network_sriov
        - console
        - restrict_devlxd
        - migration_pre_copy
        - infiniband
        - maas_network
        api_status: stable
        api_version: "1.0"
        auth: trusted
        public: false
        auth_methods:
        - tls
        environment:
          addresses: []
          architectures:
          - x86_64
          - i686
          certificate: |
            -----BEGIN CERTIFICATE-----
            MIIHhjCCBW6gAwIBAgIQTrZA/IdKStvMzx/cpZ0F8TANBgkqhkiG9w0BAQsFADA2
            ...
            rvc10de6jzB7zw==
            -----END CERTIFICATE-----
          certificate_fingerprint: 79835e469ddd6eb1c253801988b8b950bbe87e83c1d21ee8bea8ff2b6ec4df95
          driver: lxc
          driver_version: 2.1.1
          kernel: Linux
          kernel_architecture: x86_64
          kernel_version: 4.13.0-25-generic
          server: lxd
          server_pid: 15955
          server_version: "2.21"
          storage: zfs
          storage_version: 0.6.5.11-1ubuntu3

# Issue description
Friday I did a lxc conversion from dpkg to snap.  I also at one point had the edge snap installed but then backed to the beta channel.
Today I realized 'lxc list' took ~ 10 seconds for my 8 or so containers:

    $ time lxc list  | pastebinit
    http://paste.ubuntu.com/26485942/

    real	0m12.199s
    user	0m0.326s
    sys	0m10.560s

A brief description of the problem. Should include what you were
attempting to do, what you did, what happened and what you expected to
see happen.

Some debugging with brauner showed that I had two 'lxd --logfile /var/nsap/....' processes running.

    $ ps axw | grep lxd | grep "lxd --"
    15955 ?        Sl     6:21 lxd --logfile /var/snap/lxd/common/lxd/logs/lxd.log --group lxd
    22984 ?        R      0:00 lxd --logfile /var/snap/lxd/common/lxd/logs/lxd.log --group lxd

    $ ps auxf | pastebinit
    http://paste.ubuntu.com/26485852/

**Comment by stgraber** [comment]
So looking at the log, I think I figured out what's happening. You had a lxd daemon running, then the snap refreshed or the channel was changed and for whatever reason, the lxd daemon didn't properly exit, causing the service to restart and attempt to spawn a second copy of the lxd daemon.

LXD does an early check to see if the socket already exists and is already bound by lxd, if that's the case, it refuses to start. And since the systemd policy is to respawn when something fails, that lxd daemon is effectively respawned in a loop...

I'll look into adding a bit of code to the snap so that in such cases it'd kill the existing lxd daemon.
